### ISSUE TITLE
Links fail colour contrast on focus if they wrap to a new line

### ISSUE THREAD

**Issue by abbott567** [alphagov/govuk-frontend#3085]
## Description of the issue
If a link is inline, and hits the end of the container forcing it to wrap to the next line, there is no border on the bottom when it gets focused using Chrome. By only having the yellow background it would fail WCAG 2.1 on colour contrast.

<img width="738" alt="A screenshot of a link spanning multiple lines - it is highlighted in yellow." src="https://user-images.githubusercontent.com/9450064/206560922-46f6ba4d-aa5e-44ff-a5a3-a61485adc705.png">

## Steps to reproduce the issue
Go to [Typography in the GOV.UK Design System](https://design-system.service.gov.uk/styles/typography/) use the keyboard to focus the link which spans multiple lines labelled: "read the ‘If your service is not on GOV.UK’ section on ‘Making your service look like GOV.UK’"

## Actual vs expected behaviour
Expected it to have a black focus indicator like other links.

<img width="754" alt="A screenshot of a link on a single line - it is highlighted in yellow and has a thick black border on the bottom." src="https://user-images.githubusercontent.com/9450064/206561353-7a63efad-0ec4-4c6d-9e98-c8c955d96f97.png">

- Operating system: Mac OS X 12.6
- Browser: Chrome
- Browser version: 108
- GOV.UK Frontend Version: 4.4.0

**Comment by colinrotherham** [comment]
Hi @abbott567

What a fascinating bug (apparently this is "fixed" behaviour too) 🙈 

Did some research into Chromium's issue trackers

**CSS Fragmentation Module Level 3**
[5.4. Fragmented Borders and Backgrounds: the `box-decoration-break` property](https://www.w3.org/TR/css-break-3/#propdef-box-decoration-break)

**Issue 682173: -webkit-box-decoration-break doesn't cut box shadows according to the specs**
https://bugs.chromium.org/p/chromium/issues/detail?id=682173 (18 Jan 2017)

* https://github.com/web-platform-tests/wpt/pull/36538
See related [Code review comments](https://chromium-review.googlesource.com/c/chromium/src/+/3973219) from Chromium Gerrit

Looks like the fix would be for us to match the [`box-decoration-break`](https://developer.mozilla.org/en-US/docs/Web/CSS/box-decoration-break) spec too by adding:

```css
box-decoration-break: clone;
-webkit-box-decoration-break: clone;
```

There's also `-ms-box-decoration-break` but I think IE11 defaults to `clone` not `slice`

You learn something new every day!

**Comment by davidc-gds** [comment]
I've ranked this issue as 'high-severity' for now, because it appears to meet the 3 factors for a high-severity accessibility issue:

- there is reasonable evidence that it makes it difficult or impossible for some people to complete certain tasks: **for low vision users**
- it affects the accessibility of essential services or critical infrastructure: **this will affect all services which use govuk-frontend link styles**
- individual service teams cannot efficiently resolve it on their own: **it's an easy fix centrally, but would take some effort to fix across all services**

**Comment by Nooshu** [comment]
It's worth noting that this is still broken in Chrome Canary (v110)
<img width="1248" alt="Screenshot 2022-12-09 at 11 53 18" src="https://user-images.githubusercontent.com/1223960/206697473-13a70fd5-e248-4fac-9130-bc3dc52df298.png" alt="Screenshot of Chrome Canary where the issue still exists.">

**Comment by 36degrees** [comment]
Although we have a fix for this, I'm wondering if there is still a Chromium bug here – even with the default `box-decoration-break: slice` I'd expect there to be a box shadow on the bottom edge of the second line? As that's not [a 'broken edge'](https://www.w3.org/TR/css-break-3/#valdef-box-decoration-break-slice)?

The fact that it has no box shadow _at all_ seems like it might still be a bug.

**Comment by Nooshu** [comment]
> Although we have a fix for this, I'm wondering if there is still a Chromium bug here – even with the default `box-decoration-break: slice` I'd expect there to be a box shadow on the bottom edge of the second line? As that's not [a 'broken edge'](https://www.w3.org/TR/css-break-3/#valdef-box-decoration-break-slice)?
> 
> The fact that it has no box shadow _at all_ seems like it might still be a bug.

Chrome team have [requested raising a crbug](https://twitter.com/yoavweiss/status/1601930225772036096) if anyone has the time / capacity

**Comment by colinrotherham** [comment]
Thanks @Nooshu

We've raised the issue (just now) with Chromium Bugs:
https://bugs.chromium.org/p/chromium/issues/detail?id=1402736

Also added to our reported bugs repo:

* https://github.com/alphagov/reported-bugs/issues/78
